### ISSUE TITLE
Add collection parameter to functions

### ISSUE THREAD

**Issue by larkob** [ArnoldSmith86/virtualtabletop#60]
The functions FLIP, LABEL, MOVE, MOVEXY, ROTATE, SHUFFLE and SORT currently only support a holder parameter. As an outcome of the "buttons overhaul" discussion, they should additionally support the collection parameter to the effect that they can process the widgets of the collection _directly_ (rather than their children).

**Comment by ArnoldSmith86** [comment]
I'm not sure if we ruled this out in the last meeting already but maybe also add a property `ids` to all operations so you can also easily specify widgets by `id`?

**Comment by larkob** [comment]
I suggest we'd rather introduce jsonPath expression instead of adding an `id` parameter. We could simply check for `"collection": "$.[?@.id=='someID']"` and parse out the ID. At some future point, we can add the whole lot of jsonPath expressions, but right now, being able to simply work with an ad-hoc collection using a single ID would already be a huge improvement.

**Comment by ArnoldSmith86** [comment]
But isn't it one of the most common use-cases to act on widgets by ID (or an array of IDs)? Wouldn't jsonPath be pretty intimidating for that?

**Comment by 96LawDawg** [comment]
> But isn't it one of the most common use-cases to act on widgets by ID (or an array of IDs)? Wouldn't jsonPath be pretty intimidating for that?

Yes.  Very intimidating.  I'm at the limit of my ability to understand the programming we already use.  Introducing flexibility is good, but not at the expense of usability for non-programmers.  I might as well just press shift and hit number keys in random order if we have to do something like larkob's json path example. $#%%^^@NoIdeaWhatI'mDoing.

**Comment by larkob** [comment]
I don't think anyone's going to write json operations once we've got Blockly, that's why I'm not worried about jsonPath expressions.  But if this is needed already, before we get to Blockly implementation, I agree we should do it in a way that non-programmers can do it.

**Comment by RaphaelAlvez** [comment]
added in #881

**Comment by ArnoldSmith86** [comment]
That's the other way around.
